compiler/rustc_middle/src/ty/mod.rs

```diff
@@ -1934,7 +1934,7 @@ impl<'tcx> TyCtxt<'tcx> {
                     field_tys: [].into(),
                     field_names: [].into(),
                     variant_fields,
-                    variant_source_info: [source_info].into(),
+                    variant_source_info: [source_info, source_info, source_info].into(),
```

Comment:
Suggestion: this looks "sus" on a quick glance, could u leave a comment re. why it needs to be duplicated 3x? (i.e. PR description)

Reply:
No as in the code :D

Reply:
Yes, you are right. Changed to `CoroutineArgs::RESERVED_VARIANTS` (instead of fixed 3), comment added.